hoodie-client/src/main/java/com/uber/hoodie/WriteStatus.java

```diff
@@ -73,7 +73,7 @@ public WriteStatus(Boolean trackSuccessRecords, Double failureFraction) {
    * @param optionalRecordMetadata optional metadata related to data contained in {@link
    * HoodieRecord} before deflation.
    */
-  public void markSuccess(HoodieRecord record, Optional<Map<String, String>> optionalRecordMetadata) {
+  public void markSuccess(HoodieRecord record, Option<Map<String, String>> optionalRecordMetadata) {
```

Comment:
This is an API change that users need to be aware of @n3nash @bvaradar

Reply:
Conventionally, ensuring API compatibility is a critical consideration.

However, I don't think we need to worry too much at the moment. Hudi has just joined Apache and has not had a heavy historical burden. Since we decided to replace it, we should be firm. After that, we can add java.util.Optional and guava's Option as rules to the check style plugin, prohibiting users from using them. In a subsequent version, we can claim to ensure backward compatibility.

When Hudi's API is basically stable, we can mark the stability level of the API with some annotations such as `@Public`, `@PublicEvolving`, and then use tools to check their compatibility. (As far as I know, Flink tried this mechanism).